src/core/memory.c

```diff
@@ -69,5 +69,5 @@ void igraph_free(void *p) {
  */
 
 void *igraph_malloc(size_t n) {
-    return malloc(n);
+    return malloc(n + 1);
```

Comment:
Can't we make this `(n > 0) ? 1 : n`? I feel like we could be screwing things up in cases when the user attempts to allocate a chunk of memory that is exactly a power of two in order to make the area align nicely with page boundaries.

Reply:
I would not change `igraph_malloc`. Its purpose is not internal use at all, but to allow users to ensure that malloc/free are always used as a matching pair.

Reply:
I see three internal uses of `igraph_malloc` (likely added by myself in the past). I would change these to `IGRAPH_CALLOC`.